### PR TITLE
Add github actions

### PR DESCRIPTION
**Pull request by charrondev**
## Changes

- Add github actions. (PHP 7.4 -> 8.2)
- Remove travisci.
- Upgrade to PHPUnit 9 (there were a couple minor deprecations to fix).
- Remove nginx startup and use the PHP Sapi instead (greatly simplifies test setup).
  - This required removing 1 test of keepalive functionality as the PHP Sapi doesn't support it.
- Replace the jank deprecated and alpha `vanilla/garden` framework with slim for the tests.

### CODE REVIEW

File: tests/test-server.php [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+//    $request = $app->request;
+//
+//    if ($request->getEnv('PHP_AUTH_USER') !== $username) {
+//        throw new ClientException('Invalid username.', 401, ['HTTP_WWW-Authenticate' => 'Basic realm="tests"']);
+//    } elseif ($request->getEnv('PHP_AUTH_PW') !== $password) {
+//        throw new ClientException('Invalid password.', 401, ['HTTP_WWW-Authenticate' => 'Basic realm="tests"']);
+//    } else {
+//        return ['message' => 'You are in.'];
+//    }
+//});
```
Comment (vanilla-dbarbier) [inline tests/test-server.php]: Are we leaving this commented out?

**Review by vanilla-dbarbier — CHANGES_REQUESTED** [review]
Minor thing about some commented code, otherwise this makes sense to me.

File: tests/test-server.php (hunk 1 above)
Comment (charrondev) [reply tests/test-server.php]: Ah whoops, was supposed to remove this. Should be fixed now.

**Review by vanilla-dbarbier — APPROVED** [review]
*Stamp*

**Pull request merged**